ci/request-reviews/verify-base-branch.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+# Check that a PR doesn't include commits from other development branches.
+# Fails with next steps if it does
+
+set -euo pipefail
+tmp=$(mktemp -d)
+trap 'rm -rf "$tmp"' exit
+SCRIPT_DIR=$(dirname "$0")
+
+log() {
+    echo "$@" >&2
+}
+
+# Small helper to check whether an element is in a list
+# Usage: `elementIn foo "${list[@]}"`
+elementIn() {
+    local e match=$1
+    shift
+    for e; do
+        if [[ "$e" == "$match" ]]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+if (( $# < 6 )); then
+    log "Usage: $0 LOCAL_REPO HEAD_REF BASE_REPO BASE_BRANCH PR_REPO PR_BRANCH"
+    exit 1
+fi
+localRepo=$1
+headRef=$2
+baseRepo=$3
+baseBranch=$4
+prRepo=$5
+prBranch=$6
+
+# All development branches
+readarray -t devBranchPatterns < "$SCRIPT_DIR/devBranches.txt"
+git -C "$localRepo" branch --list --format "%(refname:short)" "${devBranchPatterns[@]}" > "$tmp/dev-branches"
+readarray -t devBranches < "$tmp/dev-branches"
+
+if [[ "$baseRepo" == "$prRepo" ]] && elementIn "$prBranch" "${devBranches[@]}"; then
+    log "This PR merges $prBranch into $baseBranch, no commit check necessary"
+    exit 0
+fi
```

Comment:
Spicy!

Reply:
@emilazy -- I think this implements [your most recent request](https://github.com/NixOS/nixpkgs/pull/336261#issuecomment-2398282281), if I'm simulating its effect correctly.

Reply:
Aha, perhaps it does. That would be nice, if so. Even opening and immediately self‐merging a merge conflict resolution PR would feel less dicey than pushing directly.

Reply:
Nah it doesn't because this is only the sub-script that checks whether the base branch is correct. It still requests reviews right now, but it could easily be changed by just flipping that 0 to a 1!

Reply:
Turns out it's a bit more complicated than that, because there's three different scenarios:
- Dev branch merge: Succeed but don't ping
- Wrong base branch: Fail and don't ping
- Normal: Succeed and ping

And the script would need more changes to deal with this, so I'd rather keep this out of this PR to minimise further changes.

Reply:
Yeah, that sounds reasonable. I think the case to handle specially is where the PR `HEAD` is a merge commit between commits that are all already on upstream branches.